Rename cmd from transfer-coin to transfer (#4263)
MystenLabs/sui: doc/src/contribute/cli-client.md

## Transferring coins

> Coins *are* objects yet have a specific use case that allow for native commands like transfer/merge-coin/split-coin to be used. This is different from non-coin objects that can only be mutated via [Move calls](#calling-move-code).

Reviewer: ```suggestion
Coins *are* objects, but they have a specific use case that allows you to use native commands such as `transfer`, `merge-coin`, and `split-coin`. This is different from non-coin objects that you can mutate only using [Move calls](#calling-move-code).
```
